pages/blog/a_straight_guide_to_apis_and_architecture_concepts.md

```diff
@@ -0,0 +1,102 @@
+ 
+ Then, I decided to change direction, looking for data. I thought that checking some visual and intelligible sources might clarify my head. So I could understand where to put the information, helping me start weaving a tapestry. Nevertheless, not knowing the code made it difficult for me, if not impossible, to interpret rough data. 
+
+Third attempt… I just remembered that someone said that practice makes perfect. So, the next try was in that direction. Playing an instrument is a better way to understand it than reading definitions or pentagrams. See it and touch it; feel the shape directly, the stage, the environment. It was time to take action: a colleague volunteered to guide me, practicing somehow with these concepts without prior knowledge. Just making fun, as a child, looking for references, playing, and understanding the technique while finding similarities and comparisons with real situations that anyone could understand pretty well. 
+ 
+So, let’s talk some basic concepts:
+ 
+## APIS
+
+### What is an API?
+
+APIs are program connectors. Synchronous or asynchronous, they act as glue between different applications or programs. As cables or pipes connecting different locations to facilitate information exchange or optimize resources.
+ 
+### How does an API work?
+ 
+APIs act as an intercommunicator, linking one or more programs. Let's say they could be seen as a bridge between different places connecting them and transferring the required information through them. It’s a way to exchange data or give commands to each other.
+ 
+### Types of APIs? 
+
+We can classify them into two main groups, depending on their use or their type of architecture.
+
+#### Types of APIs according to their use:
+ 
+- **Internal APIs** are those used at a local level. They focus on communicating with each other within the same system or the same computer. An excellent example of them might be the pipes contained in a house. They connect the sink to the bathroom and the washing machine to the dryer, for example.
+ 
+- **External APIs**, or web APIs, are those used to communicate two or more applications over the Internet. We could exemplify it with the pipes that connect different houses in the same block of apartments or neighborhoods.
+
+- **Partner APIs** are a middle ground between internal and external APIs. This is because they are accessible to people outside the organization but only to those with exclusive permissions. Typically, this special access is granted to certain third parties to facilitate collaboration. 
+
+  A good example might be when we invite friends over, and they bring their controllers to play the game console. Or when a friend comes over and connects to our wifi network.
+
+#### According to its **architectural style**: 
```

Comment:
```suggestion
#### According to its architectural style: 
```

bold will not render in TOC and will show asterix

![Screenshot 2022-12-05 at 12 36 31](https://user-images.githubusercontent.com/6995927/205627893-c3fe4b49-2274-4963-a7e1-9490ff6f2858.png)

Reply:
you probably should remove `:` too as it looks weird in TOC